google-cloud-storage/src/main/java/com/google/cloud/storage/WriteFlushStrategy.java

```diff
@@ -121,6 +146,27 @@ private static WriteObjectRequest possiblyPairDownRequest(
     return b.build();
   }
 
+  private static BidiWriteObjectRequest possiblyPairDownBidiRequest(
+      BidiWriteObjectRequest message, boolean firstMessageOfStream) {
+    if (firstMessageOfStream && message.getWriteOffset() == 0) {
+      return message;
+    }
+
+    BidiWriteObjectRequest.Builder b = message.toBuilder();
+    if (!firstMessageOfStream) {
+      b.clearUploadId();
+    }
+
+    if (message.getWriteOffset() > 0) {
+      b.clearWriteObjectSpec();
+    }
+
+    if (message.getWriteOffset() > 0 && !message.getFinishWrite()) {
+      b.clearObjectChecksums();
```

Comment:
Does this remove checksums for every message after initial / ignoring last? I would expect checksums to be supplied for intermediate messages as well.

Reply:
Ignore my understanding was incomplete.